### ISSUE TITLE
[CircularProgressIndicator] CircularIndeterminateAnimatorDelegate causes NPE 

### ISSUE THREAD

**Issue by dniHze** [material-components/material-components-android#2251]
**Description:** When updating CircularProgressIndicator progress with `animated = true` param there is a chance if CPI is detached from window (for example on fragment detach), `completeEndAnimator` will cause NullPointerException.

Stacktrace:
```java
java.lang.NullPointerException: Attempt to invoke virtual method 'void t.a0.a.a.b.a(android.graphics.drawable.Drawable)' on a null object reference
    at com.google.android.material.progressindicator.CircularIndeterminateAnimatorDelegate$2.onAnimationEnd(CircularIndeterminateAnimatorDelegate.java:112)
    at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:552)
    at android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1232)
    at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1474)
    at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
    at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
    at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1075)
    at android.view.Choreographer.doCallbacks(Choreographer.java:889)
    at android.view.Choreographer.doFrame(Choreographer.java:818)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1063)
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:233)
    at android.app.ActivityThread.main(ActivityThread.java:7225)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:499)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:962)
```

It seems like the view removes the requested listener on detaching from a window, but the `completeEndAnimator` indeterminate animator is not canceled, so it finishes and trying to call `null`-ed listener.
  
**Expected behavior:** CPI does not cause exceptions when detached from a window.

**Source code:** 
```kotlin
// Somewhere in fragment update CPI 
binding.circularProgresssIndicator.isIndeterminate = true

// Then migrate CPI to determinate state with animation
binding.circularProgressIndicator.setProgressCompat(value, true)
// This causes completeEndAnimator to start

// Detach fragment midprogress for some reason (for example: dialog is no longer actual)
// This leads to unregisterAnimatorsCompleteCallback called on CircularIndeterminateAnimatorDelegate

// completeEndAnimator called on onAnimationEnd, that leads to NPE on animatorCompleteCallback.

NPE raised 💥
```

**Android API version:** 28 - 30

**Material Library version:** 1.4.0-rc01

**Device:** Redmin Note 9 Pro, Mi 8 Lite

To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by dniHze** [comment]
I'm OK with submitting PR and add callback null check in`CircularIndeterminateAnimatorDelegate`, but I'm not sure if this going to break some logic behind.

**Comment by pekingme** [comment]
Fixed with f732070619797d8af5fa5ac0ab535ad8dd59d420

**Comment by Jaydipnathani** [comment]
Which Version fix this issue ?

**Comment by Jaydipnathani** [comment]
> I'm OK with submitting PR and add callback null check in`CircularIndeterminateAnimatorDelegate`, but I'm not sure if this going to break some logic behind.



> Fixed with [f732070](https://github.com/material-components/material-components-android/commit/f732070619797d8af5fa5ac0ab535ad8dd59d420)

Which Version fix this issue ?

**Comment by drchen** [comment]
A new version will be released in a couple of days, which will include the fix. 

Please check out on the release page:
https://github.com/material-components/material-components-android/releases
